src/nvim/extmark.c

```diff
@@ -513,7 +555,6 @@ void extmark_adjust(buf_T *buf, linenr_T line1, linenr_T line2, linenr_T amount,
     old_row = line2 - line1 + 1;
```

Comment:
Hmm, is this `+ 1` correct?
1. Place 2 marks on line 1 and line 2 with `invalidate == true`.
2. Delete the first line
3. `extmark_splice_impl()` is called with old_row == 1
4. Both the mark on line 1 and line 2 are removed. The one on line 2 shouldn't be.

Reply:
I guess it is done to propagate line deletion to deeper functions?
Maybe the loop `break` in `extmark_splice_delete()` should be adjusted by 1.

Reply:
I've added a condition that fixes this without breaking tests, but doesn't really make sense in my eyes. I don't see why this one-off error cannot be adjusted in the `u_row` argument to `extmark_splice_delete()` without breaking existing undo/redo tests. Seems to me it is looping one row too far, so I added adjusted the invalidation condition to at least not remove marks that shouldn't be.

Reply:
yep, it is for marks that needs to be adjusted on the line after, so making it narrower in the deletion loop makes sense.